src/containers/Application/ApplicationPage.tsx

```diff
@@ -2,32 +2,96 @@ import React, { useEffect, useState } from 'react'
 import { FullStructure, ResponsesByCode, ElementStateNEW } from '../../utils/types'
 import useGetFullApplicationStructure from '../../utils/hooks/useGetFullApplicationStructure'
 import { ApplicationStatus } from '../../utils/generated/graphql'
-import { useApplicationState } from '../../contexts/ApplicationState'
 import { useUserState } from '../../contexts/UserState'
 import { useRouter } from '../../utils/hooks/useRouter'
-import { Loading, NoMatch } from '../../components'
+import { Loading } from '../../components'
 import strings from '../../utils/constants'
-import messages from '../../utils/messages'
 import { Button, Grid, Header, Message, Segment, Sticky } from 'semantic-ui-react'
 import { PageElements } from '../../components/Application'
+import { useFormElementUpdateTracker } from '../../contexts/FormElementUpdateTrackerState'
 
 interface ApplicationProps {
   structure: FullStructure
   responses?: ResponsesByCode
 }
+// TODO move to shared types if used elsewhere
+type SectionAndPage = { sectionCode: string; pageName: string } | null
```

Comment:
I'm using a very similar one in ProgressBar PR : )
https://github.com/openmsupply/application-manager-web-app/blob/b55cf0fd25c334ae09ac5207f34c53225b8a53e4/src/components/Application/ProgressBarNEW.tsx#L9-L12

Reply:
Also I don't think this needs to be a type and set to either value | null. You can just set the value here and in the state holding this `strictSectionPage` it is set to `undefined` until you have some `SectionAndPage` to define. The state can be either `SectionAndPage` | undefined - without having to define that

Reply:
Doesn't it need to be specifically `null`, returning 'undefined' is pretty weird (an empty return statement), likewise setting things to 'undefined'

getFirstInvalidPage needs to return an indication that there are no invalid pages and pass that through to MethodToCallOnRevalidation, I thought null would be the way to go (and the type is sharable).

To use this type in other places, yeah it's better to just have it ` sectionCode: string; pageName: string }`
1b280a6189c93cfc82302916d5cfd682866f817c

Reply:
That's all good. As long as it makes sense to us, I feel that using `undefined` for that is also good. But we need to keep the same pattern everywhere. Otherwise is confusing.